### PR TITLE
Add UT for pod toleration

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

Add UT for pod toleration

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by kevin-wangzefeng** [comment]
@wackxu 

Thanks, can you also update the description in this test (L59 of toleration_test.go)
```go
 		{
 			description: "toleration and taint have the same effect, toleration has empty key and operator is Exists, means match all taints, expect tolerated",
 			toleration: Toleration{
 				Key:      "",
 				Operator: TolerationOpExists,
 				Effect:   TaintEffectNoSchedule,
 			},
 			taint: Taint{
 				Key:    "foo",
 				Value:  "bar",
 				Effect: TaintEffectNoSchedule,
 			},
 			expectTolerated: true,
 		},
```
"means match all taints" should be "match all the taints that have the same effect with given toleration.Effect"

**Comment by wackxu** [comment]
@kevin-wangzefeng Thanks for your review. And I have  fix it.

**Comment by wackxu** [comment]
@smarterclayton @saad-ali @nikhiljindal  Please take a look at this PR, it needs your approve, thanks.

**Comment by wackxu** [comment]
@smarterclayton @saad-ali @nikhiljindal Please take a look at this PR, it needs your approve, thanks.

File: staging/src/k8s.io/api/core/v1/toleration_test.go [hunk 1]
```diff
@@ -114,6 +114,20 @@ func TestTolerationToleratesTaint(t *testing.T) {
 			},
 			expectTolerated: false,
 		},
+		{
+			description: "toleration have an empty key with operator Exists, expect tolerate everything",
```
Comment (bsalamat) [inline staging/src/k8s.io/api/core/v1/toleration_test.go]: s/have an empty key with operator Exists/have an empty Key and empty Effect with operator Exists/

Comment (wackxu) [reply staging/src/k8s.io/api/core/v1/toleration_test.go]: Done

File: staging/src/k8s.io/api/core/v1/toleration_test.go [hunk 2]
```diff
@@ -114,6 +114,20 @@ func TestTolerationToleratesTaint(t *testing.T) {
 			},
 			expectTolerated: false,
 		},
+		{
+			description: "toleration have an empty key and empty Effect with operator Exists, expect tolerate everything",
```
Comment (bsalamat) [inline staging/src/k8s.io/api/core/v1/toleration_test.go]: Sorry for not saying it the first time. Could you please change the description to:
"toleration with an empty Key, empty Effect, and operator Exists, should tolerate all taints"

**Review by bsalamat — COMMENTED** [review]
Just a minor comment, otherwise LGTM.

File: staging/src/k8s.io/api/core/v1/toleration_test.go (hunk 2 above)
Comment (wackxu) [reply staging/src/k8s.io/api/core/v1/toleration_test.go]: OK, no problem, I have fix it.

**Review by bsalamat — COMMENTED** [review]
/lgtm

(you need an approver of API to approve this)

**Comment by wackxu** [comment]
@smarterclayton @saad-ali @nikhiljindal Please take a look at this PR, it needs your approve, thanks.

**Comment by wackxu** [comment]
kindly ping @smarterclayton for approval

**Comment by bsalamat** [comment]
ping @smarterclayton for approval

**Comment by bsalamat** [comment]
/remove-lifecycle stale

ping @kubernetes/api-reviewers

**Pull request closed without merging**